core/src/main/java/com/predic8/membrane/core/interceptor/rewrite/ReverseProxyingInterceptor.java

```diff
@@ -90,8 +107,12 @@ public Outcome handleResponse(Exchange exc) throws Exception {
 		if (exc.getOriginalHostHeaderHost() == null) {
 			// just remove the schema/hostname/port. this is illegal (by the spec),
 			// but most clients understand it
-			exc.getResponse().getHeader().setValue(LOCATION, new URL(location).getFile());
-			return CONTINUE;
+            try {
+                exc.getResponse().getHeader().setValue(LOCATION, new URL(location).getFile());
+            } catch (MalformedURLException e) {
+				log.error("Could not parse URL for Location header: {}",location, e);
```

Comment:
this would `CONTINUE` the flow with an illegal URL in the `Location` HTTP header. probably it shoudn't.